src/win.rs

```diff
@@ -769,11 +769,7 @@ impl Win {
     }
 
     fn draw_cells(&self, cs: &[char], prop: GlyphProp, xp: usize, yp: usize) {
-        let GlyphProp { mut fg, bg, attr } = prop;
-        if attr.contains(GlyphAttr::BOLD) && fg < 8 {
-            fg += 8;
-        }
-
```

Comment:
this makes the blue color (e.g. "ls" output for directory) not easily readable.

Reply:
Interesting, what happens with 'real' st?  I think st tries to use the bold font and leave the color alone with some hardcoded color if there is no bold (but would need to double check that).  I do not think rterm handles a missing bold font yet.  My color ls does not seem that bad (dark blue for dirs on black) with the default colors but it is not a great combo and probably depends on the screen (I tested on a bright laptop screen).  It feels wrong to just 'brighten' all the colors though if ls picked a bad scheme or we have a bad background color?  Trying xterm, it defaults to a white background so that is probably why ls uses blue...  Since I completely change the first 16 colors to the solarized colors and they do not follow the 8 color plus 8 bright versions it really messing me up (I assume other color schemes will have the same problem).  Also, normal st does not have an issue with my scheme.  We could change the default background color, maybe black is not a good call there.  White backgrounds drive me nuts but maybe something darker but not black?  If you want to play with backgrounds you can send an OSC 11 code to try some out.  I am not sure my defaults would be a good call, I like them, but they may be a bit unexpected for a default scheme.  The more I think about this the more I think maybe the black background is the issue?

I saw your note on making me a maintainer, thanks.  Once it looks like rterm can replace st I was going to open an issue to discuss where you wanted to go with rterm.  I have some stuff I want to do but it will be getting a further afield then st.  I will still do that, I can fork and keep rterm as a base if you do not see it going in some of the directions I may move into but I will open an issue to discuss before then.

Do you mind if I add a LICENCE file (MIT/Apache per the Cargo.toml)?  Seems like a good idea to let potential users know upfront what the terms are (I initially did not catch it in Cargo.toml was not sure what if any license rterm had).  I might also add some high level coverage of supported escape codes to the readme if you are cool with that.  Both of these things are not fun, so not in a hurry, but I will probably do them if you are cool with it.

Probably won't be able to maintain my pace from this week (was on vacation for one) and have been neglecting some other projects but hacking on rterm was fun and it will intersect with my other open source stuff soon.  Also, I am moving to using it under tmux instead of st so I will have an interest in rterms maintenance...

Reply:
Yes, feel free to add a LICENSE file and anything else you would like to change.

I'll probably do some small refactors without changing the behavior of the code.  I am open to new features as long as the code is well organized.  I am also relatively new to rust, so please don't hesitate to give your opinion on coding styles, etc.

Here's the reference st code for the bold color conversion: https://git.suckless.org/st/file/x.c.html#l1386

Reply:
Roger, if I get really motivated I may add some docs and tests (at least to the public term functions) but I may not get that motivated...  Tests are obviously tricky with a terminal.  Maybe a github action to test fmt/clippy/compile and later run tests.

I was thinking it might make sense to make win a trait and add an xwin for the x implementation.  That way someone could add an OSX, Win (would that even make sense?), OpenGL (all the cool kids seem to need GPU rendered terms these days) or whatever backend easily (or maybe even a terminal backed win for a multiplexer).  That should be real easy though with how you set things up.

I think you have organized things great, I have found it easy to follow and have not felt frustrated by the organization at all.  So great job, especially if you are new to rust (and diving straight into C FFI which is not super fun in Rust).  Hopefully I did not steal your fun if this was a learning project...

I will restore the fg + 8 stuff.  You are obviously correct about st which made me try to figure out how I have been using st for years and not had the problem I was having.  Turns out I flipped two of my palette entries by mistake (using a script that uses OSC codes to set the palette) which caused bold red to be background instead of orange...  It is a little funky when your palette does not follow the 8-15 are brighter convention but is fine if set up correctly (and obviously I have been using it for a long time and not bothered to investigate- I did notice that sometimes a color seemed odd like grey for dirs in ls or ripgrep using orange but it was readable so who cares).  The solarized colors all basically work together anyway.  I also tested xterm for kicks and it does the same thing.  Sorry about the confusion and thanks for catching it.